src/config.rs

```diff
@@ -124,17 +130,34 @@ impl TestConfig {
     ///
     /// If you don't want to change any of the defaults, you can use `TestConfig::new` to construct
     /// a TestConfig with the default keywords (which are listed in its documentation).
-    pub fn with_custom_keywords(binary_path: &str, test_path: &str, test_line_prefix: &str,
+    pub fn with_custom_keywords<Binary, Tests>(binary_path: Binary, test_path: Tests, test_line_prefix: &str,
                                 test_args_prefix: &str, test_stdout_prefix: &str,
-                                test_stderr_prefix: &str, test_exit_status_prefix: &str) -> TestConfig {
-        TestConfig {
-            binary_path:             binary_path.to_string(),
-            test_path:               PathBuf::from(test_path),
+                                test_stderr_prefix: &str, test_exit_status_prefix: &str) -> TestResult<TestConfig>
+    where
+        Binary: Into<PathBuf>,
+        Tests: Into<PathBuf>,
+    {
+        let (binary_path, test_path) = (binary_path.into(), test_path.into());
+
+        if !test_path.exists() {
+            println!("{}", format!("the given test path '{}' does not exist", test_path.display()).red());
+
+            return Err(TestError::MissingTests(test_path));
+        } else if !test_path.is_dir() {
+            println!("{}", format!("the given test path '{}' is not a directory", test_path.display()).red());
+
+            return Err(TestError::ExpectedDirectory(test_path));
+        }
+
+
+        Ok(TestConfig {
```

Comment:
Can we merge this `Ok` into an else branch of the if statements above so we don't need early returns?